x-pack/plugins/reporting/export_types/printable_pdf/server/execute_job/compatibility_shim.js

```diff
@@ -27,11 +39,49 @@ export function compatibilityShimFactory(server) {
     throw new Error(`Unable to generate report for url ${savedObject.url}, it's not a Kibana URL`);
   };
 
+  const getSerializedSession = async (decryptedHeaders, jobSession) => {
+    if (!server.plugins.security) {
+      return null;
+    }
+
+    if (jobSession) {
+      try {
+        return await crypto.decrypt(jobSession);
+      } catch (err) {
+        throw new Error('Failed to decrypt report job data. Please re-generate this report.');
+      }
+    }
+
+    const cookies = decryptedHeaders.cookie ? cookie.parse(decryptedHeaders.cookie) : null;
+    if (cookies === null) {
+      return null;
+    }
+
+    const cookieName = server.plugins.security.getSessionCookieOptions().name;
+    if (!cookieName) {
+      throw new Error('Unable to determine the session cookie name');
+    }
+
+    return cookies[cookieName];
+  };
+
   return function (executeJob) {
     return async function (job, cancellationToken) {
       const urls = job.objects.map(savedObject => getSavedObjectAbsoluteUrl(job, savedObject));
+      const decryptedHeaders = await decryptJobHeaders(job);
+      const authorizationHeader = decryptedHeaders.authorization;
```

Comment:
Honest question: are we guaranteed that `decryptedHeaders` will be defined?

Reply:
The headers were always stored, even if they were null, so worst-case it should be an empty object